source/dub/packagemanager.d

```diff
@@ -533,7 +533,18 @@ class PackageManager {
 				try foreach( pdir; iterateDirectory(path) ){
 					logDebug("iterating dir %s entry %s", path.toNativeString(), pdir.name);
 					if( !pdir.isDirectory ) continue;
-					auto pack_path = path ~ (pdir.name ~ "/");
+
+					// Search for a single directory within this directory which happen to be a prefix of pdir
+					// This is to support new folder structure installed over the ancient one.
+					FileInfo subdir;
+					foreach( pdir2; iterateDirectory(path ~ (pdir.name ~ "/")) )
+					{
+						if (pdir.name.startsWith(pdir2.name)) // eg: package vibe-d will be in "vibe-d-x.y.z/vibe-d"
+							subdir = pdir2;
+					}
+					if( !subdir.isDirectory ) continue;
```

Comment:
I would rather just upgrade the existing structure once and directly load subpaths here than supporting 2 different layouts.
You can place a `.new_package_layout` dummy in the dub folder to mark it as converted.